src/network/serverpackethandler.cpp

```diff
@@ -471,6 +471,12 @@ void Server::process_PlayerPos(RemotePlayer *player, PlayerSAO *playersao,
 	v3f position((f32)ps.X / 100.0f, (f32)ps.Y / 100.0f, (f32)ps.Z / 100.0f);
 	v3f speed((f32)ss.X / 100.0f, (f32)ss.Y / 100.0f, (f32)ss.Z / 100.0f);
 
+	const v3s16 old_node_position = floatToInt(playersao->getBasePosition(), BS);
```

Comment:
I'm not sure this will get the old position.
This line is unnecessary as you only need to detect a change of the rounded position.